scipy/interpolate/tests/test_bsplines.py

```diff
@@ -2332,6 +2354,36 @@ def test_design_matrix(self):
         with assert_raises(ValueError, match="Data and knots*"):
             NdBSpline.design_matrix([[1, 2]], t3, [k]*3)
 
+    def test_concurrency(self):
+        barrier = threading.Barrier(10)
```

Comment:
1. What is the 10 argument?

2. Some added tests have a `threading.Barrier`, some do not. What is the reasoning, when does one need it or not?  Ideally, there are some high-level explanations somewhere in the devguide (for those of us who only ramp up to threading), and specific comments in added tests.

Reply:
1. A barrier is a synchronization mechanism that ensures that multiple threads reach the same point of execution, and once all threads reach the barrier, they continue the execution. In this case, the number 10 refers to the number of threads that are expected to wait on the barrier, this number happens to be the same as the number of threads spawned later on
2. I started adding them later on in order to better enforce concurrency. i.e., make sure that all threads call the same function at the same time